forui/lib/src/widgets/slider/slider_controller.dart

```diff
@@ -0,0 +1,207 @@
+import 'dart:collection';
+
+import 'package:flutter/foundation.dart';
+import 'package:forui/forui.dart';
+import 'package:forui/src/foundation/platform.dart';
+import 'package:forui/src/widgets/slider/slider_selection.dart';
+
+/// Possible ways for a user to interact with a slider.
+enum FSliderInteraction {
+  /// Allows the user to interact with the slider by sliding anywhere on the track.
+  slide,
+
+  /// Allows the user to interact with the slider by sliding only the slider thumb.
+  slideThumb,
+
+  /// Allows the user to interact with the slider by tapping and sliding the slider thumb.
+  tapAndSlideThumb,
+
+  /// Allows the user to interact with the slider by tapping anywhere.
+  tap,
+}
+
+/// A controller that manages a slider's active track.
+///
+/// This class should be extended to customize selection. By default, the following controllers are provided:
+/// * [FContinuousSliderController.new] for selecting a single continuous value.
+/// * [FContinuousSliderController.range] for selecting continuous range.
+/// * [FDiscreteSliderController.new] for selecting a discrete value.
+/// * [FDiscreteSliderController.range] for selecting a discrete range.
+abstract class FSliderController extends ChangeNotifier {
+  static FSliderInteraction get _platform =>
+      touchPlatforms.contains(defaultTargetPlatform) ? FSliderInteraction.slide : FSliderInteraction.tapAndSlideThumb;
+
+  /// True if the registered tooltip(s) should be shown when the user interacts with the slider. Defaults to true.
+  final FSliderTooltipsController tooltips;
+
+  /// The allowed ways to interaction with the slider.
+  ///
+  /// Single-value sliders default to [FSliderInteraction.slide] on Android, Fuchsia and iOS, and
+  /// [FSliderInteraction.tapAndSlideThumb] on other platforms.
+  ///
+  /// Range sliders always default to [FSliderInteraction.tapAndSlideThumb].
+  final FSliderInteraction allowedInteraction;
+
+  /// Whether the active track is extendable at its min and max edges.
```

Comment:
What exactly does extendable mean?

Reply:
Whether the active track is extendable (growable) at the min/max edges. I opted to use extendable since used "extent" throughout the slider.

Reply:
I'm honestly a bit confused, isn't the track length fixed?

Reply:
Ah maybe the name isn’t the best but it refers to the active track/selection.

Reply:
Oh, so if it's true, a thumb will be present for that edge?

Reply:
Yup!